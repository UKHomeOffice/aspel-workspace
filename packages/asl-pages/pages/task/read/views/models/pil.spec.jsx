import React from 'react';
import {shallow} from 'enzyme';
import Pil from "./pil";

describe('<PIL />', () => {

  test('renders when action is update', () => {
    const props = {
      task: {
        data: {
          data: {
            certificates: [{}, {}]
          }
        },
        isOpen: false
      }
    };
    const component = shallow(<Pil {...props} />);
    expect(component).toMatchSnapshot();
  });

  test('renders only conditions when action is update-conditions', () => {
    const props = {
      task: {
        data: {
          action: 'update-conditions',
          data: {
            conditions: 'old-conditions'
          }
        },
        isOpen: false
      },
      values: {
        conditions: 'new-conditions'
      }
    };
    const component = shallow(<Pil {...props} />);
    expect(component).toMatchSnapshot();
  });
});
