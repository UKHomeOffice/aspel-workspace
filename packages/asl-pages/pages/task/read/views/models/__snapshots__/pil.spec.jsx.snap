// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PIL /> renders only conditions when action is update-conditions 1`] = `
<ForwardRef
  id="conditions"
  key="conditions"
>
  <h2>
    <Connect(Snippet)>
      sticky-nav.conditions
    </Connect(Snippet)>
  </h2>
  <DiffText
    currentLabel={
      <UNDEFINED>
        diff.previous
      </UNDEFINED>
    }
    newValue="old-conditions"
    oldValue="new-conditions"
    proposedLabel={
      <UNDEFINED>
        diff.changed-to
      </UNDEFINED>
    }
  />
</ForwardRef>
`;

exports[`<PIL /> renders when action is update 1`] = `
Array [
  <ForwardRef
    id="procedures"
    key="procedures"
  >
    <h2>
      <Connect(Snippet)>
        sticky-nav.procedures
      </Connect(Snippet)>
    </h2>
    <PilProcedures
      task={
        Object {
          "data": Object {
            "data": Object {
              "certificates": Array [
                Object {},
                Object {},
              ],
            },
          },
          "isOpen": false,
        }
      }
    />
  </ForwardRef>,
  <ForwardRef
    id="species"
    key="species"
  >
    <h2>
      <Connect(Snippet)>
        sticky-nav.species
      </Connect(Snippet)>
    </h2>
    <SpeciesDiff
      after={
        Object {
          "certificates": Array [
            Object {},
            Object {},
          ],
        }
      }
    />
  </ForwardRef>,
  <ForwardRef
    id="training"
    key="training"
  >
    <h2>
      <Connect(Snippet)>
        sticky-nav.training
      </Connect(Snippet)>
    </h2>
    <SummaryTable
      certificates={Array []}
    />
  </ForwardRef>,
]
`;
